### ISSUE TITLE
can't create an image using build_image.sh

### ISSUE THREAD

**Issue by IdoGabay111** [pytorch/serve#1214]
## Context

* torchserve version: ?
* torch-model-archiver version: ?
* torch version: 1.9.0
* torchvision version [if any]: 0.10.0
* torchtext version [if any]: -
* torchaudio version [if any]: -
* java version: -
* Operating System and version: macOS 11.5.2

## Your Environment

* Installed using source? [yes/no]: yes
* Are you planning to deploy it using docker container? [yes/no]: yes
* Is it a CPU or GPU environment?: cpu
* Provide config.properties, logs [ts.log] and parameters used for model registration/update APIs:

## Expected Behavior

creating an image using build_image.sh as described on the readme file in the docker folder

## Current Behavior

while running the script i get an error : "No matching distribution found for torch==1.9.0+cpu"
and the image in not created.

## Possible Solution

might be something related to python version or support of M1 chip. 

## Failure Logs [if any]

<img width="1440" alt="Screen Shot 2021-08-25 at 15 54 26" src="https://user-images.githubusercontent.com/89010709/130794495-ac8a911a-35ee-483f-abf6-e0eba31baae7.png">

**Comment by lxning** [comment]
I was not able to reproduce this error.
[log.txt](https://github.com/pytorch/serve/files/7094701/log.txt)

@IdoGabay111 could you run python to see if your sys platform is darwin? eg.
python
Python 3.9.6 (default, Jun 29 2021, 06:20:32)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.platform
'darwin'

**Comment by IdoGabay111** [comment]
@lxning 
the output i get is : 
(base) ido@Idos-MacBook-Air ~ % python
Python 3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:35:11)
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.

**Comment by lxning** [comment]
@IdoGabay111 
Can you run the following code in python?

import sys
sys.platform

**Comment by IdoGabay111** [comment]
@lxning 
for printing sys.platform line i get the output "darwin"

**Comment by lxning** [comment]
@IdoGabay111 according to [code](https://github.com/pytorch/serve/blob/master/ts_scripts/install_dependencies.py), [torch_darwin.txt](https://github.com/pytorch/serve/blob/master/requirements/torch_darwin.txt) should be called if platform is "darwin".  That means "torch==1.9.0" will be installed, not "torch==1.9.0+cpu" (shown on your log).

I tested on my mac, it works fine (see my previous attached log). Can you run "git pull" to sync up your code with master or branch release_0.4.2 and then call build_image.sh?

**Comment by IdoGabay111** [comment]
@lxning I did it.... doesn't work.

I manually tried to delete "+cpu" in the Dockerfile. now the script can install torch!
unfortunately now it can't find the torchvision version :

<img width="1440" alt="Screen Shot 2021-09-09 at 16 27 56" src="https://user-images.githubusercontent.com/89010709/132694742-c0f58760-249f-4640-a61d-76e3a8233b53.png">


So i tried to delete "+cpu" there as well but sadly it didn't work :

<img width="1440" alt="Screen Shot 2021-09-09 at 16 26 42" src="https://user-images.githubusercontent.com/89010709/132694981-75e56d14-2bfa-46d2-8710-7b0fa89c30dc.png">
